src/lib/core/errors.ts

```diff
@@ -39,15 +40,52 @@ export function isConsentRequiredError(test: unknown): test is ConsentRequiredEr
   );
 }
 
+/**
+ * An error that includes an `authorization_parameters` property, a.k.a "G.A.R.E".
+ *
+ * A well-known error shape is provided by services when additional authorization requirements must be met by the session.
+ * This object can be converted to parameters accepted by Globus Auth using `sdk.errors.toAuthorizationQueryParams()`.
+ */
 export type AuthorizationRequirementsError = {
   authorization_parameters: {
-    session_message: string;
-    session_required_identities: string[];
-    session_required_mfa: boolean;
-    session_required_single_domain: string[];
+    session_message?: string;
+    session_required_identities?: string[];
+    session_required_mfa?: boolean;
+    session_required_single_domain?: string[];
+    prompt?: string;
+    required_scopes?: string[];
   };
   [key: string]: unknown;
 };
+/**
+ * Keys that should not be included in the query string object (not recognized by Globus Auth).
+ */
+const NO_OP_KEYS: (keyof AuthorizationRequirementsError)[] = ['required_scopes'];
+/**
+ * Convert an `AuthorizationRequirementsError` to a query string object accepted by Globus Auth.
+ */
+export function toAuthorizationQueryParams(
+  error: AuthorizationRequirementsError,
+): AuthorizationQueryParameters {
+  return Object.entries(error.authorization_parameters).reduce((acc, [key, v]) => {
```

Comment:
Can this be done with `URLSearchParams`?

Reply:
I gave it a go thinking it would at least take care of the `value` conversion, but it looks like Typescript forces you to use a `string`( https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.urlsearchparams.html#set), even though the browser seems to convert properly...

<img width="360" alt="Screenshot 2024-06-12 at 4 09 31 PM" src="https://github.com/globus/globus-sdk-javascript/assets/694253/dbab6a0d-aed1-4ddc-bd0b-1a04c50be961">

Either way, even if that cleaned up some of this, there would still need to be conversion from `URLSearchParams` back to a POJO, so I think just sticking with the  with the object parsing for now is ok.